File: src/main/java/me/taubsie/dungeonhub/kord/application/misc/Leaderboard.java
package me.taubsie.dungeonhub.kord.application.misc;

import dev.kord.rest.builder.message.EmbedBuilder;
import lombok.Getter;
import me.taubsie.dungeonhub.kord.application.service.LeaderboardService;
import me.taubsie.dungeonhub.common.model.score.LeaderboardModel;

//TODO migrate to kotlin
/**
 * This class holds the data for a simple leaderboard without pages.
 * It contains the title of the leaderboard and the score data, to allow easy querying and building.
 */
public class Leaderboard {
    private final String title;
    @Getter
    private final LeaderboardModel leaderboardModel;

    public Leaderboard(String title, LeaderboardModel leaderboardModel) {
        this.title = title;
        this.leaderboardModel = leaderboardModel;
    }

    public String getLeaderboardTitle() {
        return title;
    }

    public boolean isEmpty() {
        return leaderboardModel == null || leaderboardModel.getScores().isEmpty();
    }

    public EmbedBuilder getEmbed() {
        if (isEmpty()) {
            return LeaderboardService.INSTANCE.getEmptyLeaderboardEmbed(title);
        }

        return LeaderboardService.INSTANCE.getLeaderboardEmbed(title, leaderboardModel);
    }
}

File: src/main/java/me/taubsie/dungeonhub/kord/application/misc/Leaderboard.kt
package me.taubsie.dungeonhub.kord.application.misc

import dev.kord.rest.builder.message.EmbedBuilder
import me.taubsie.dungeonhub.common.model.score.LeaderboardModel
import me.taubsie.dungeonhub.kord.application.service.LeaderboardService.getEmptyLeaderboardEmbed
import me.taubsie.dungeonhub.kord.application.service.LeaderboardService.getLeaderboardEmbed

/**
 * This class holds the data for a simple leaderboard without pages.
 * It contains the title of the leaderboard and the score data, to allow easy querying and building.
 */
class Leaderboard(private val leaderboardTitle: String, private val leaderboardModel: LeaderboardModel?) {
    val isEmpty: Boolean
        get() = leaderboardModel == null || leaderboardModel.scores.isEmpty()

    val embed: EmbedBuilder
        get() {
            if (isEmpty) {
                return getEmptyLeaderboardEmbed(leaderboardTitle)
            }

            return getLeaderboardEmbed(leaderboardTitle, leaderboardModel)
        }
}